client-core/src/wallet/syncer.rs

```diff
@@ -484,7 +485,7 @@ pub(crate) struct FilteredBlock {
     /// Block time
     pub block_time: Time,
     /// List of successfully committed transaction ids in this block and their fees
-    pub valid_transaction_fees: BTreeMap<TxId, Fee>,
+    pub valid_transaction_fees: IndexMap<TxId, Fee>,
```

Comment:
how about
Vec< (TxId, Fee) >
because order should be preserved

Reply:
index map is actually
IndexMap == Hashtable + List

if we port to other language (haskell, c),
it would much simpler if we use Vec

Reply:
changing to `IndexMap` seems to be the simplest solution to #1069 -- changing to Vec (or other restructuring) would require more complex code changes